src/care/array_utils.h

```diff
@@ -537,9 +541,9 @@ inline void IntersectArrays(RAJA::seq_exec exec,
  ************************************************************************/
 
 template <typename T>
-inline int BinarySearch(const T *map, const int start,
-                        const int mapSize, const T num,
-                        bool returnUpperBound)
+CARE_HOST_DEVICE inline int BinarySearch(const T *map, const int start,
```

Comment:
Is checkSorted also CARE_HOST_DEVICE?

Reply:
I made it CARE_HOST_DEVICE, should be fine